src/engine/e_bluefield.c

```diff
@@ -45,24 +48,35 @@ static int engine_pka_init(ENGINE *e);
 static int engine_pka_finish(ENGINE *e);
 void engine_load_pka_int(void);
 
-/* BN operations */
 /* BN mod_exp */
 static int engine_pka_bn_mod_exp(BIGNUM *r, const BIGNUM *a, const BIGNUM *p,
                                  const BIGNUM *m, BN_CTX *ctx, BN_MONT_CTX *m_ctx);
-/* BN mod_inv */
-static int
-engine_pka_bn_mod_inv(const EC_GROUP *group, BIGNUM *r, const BIGNUM *x,
-                      BN_CTX *ctx);
 
-# if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
+/* DSA stuff */
+static int engine_pka_dsa_init(DSA *dsa);
+static int engine_pka_dsa_finish(DSA *dsa);
+
 /* RSA stuff */
-#ifndef NO_RSA
```

Comment:
same question?

Reply:
And what about this one? My understanding is OpenSSL has various build options. What if RSA config is excluded?

Reply:
Yes, right. I'll add this back.

Reply:
I have added the guards, Thanks.